server/NewBankClientHandler.java

```diff
@@ -118,4 +147,41 @@ public String printCustomerMenu () {
 		out.println("+---------------------------------------------------+");
 		return "\n";
 	}
+
+	//counts the frequency of usernames for which logins have been attempted and checks against maximum attempts set
+	private Map<String, Integer> userLogInFrequency(){
+		Map<String, Integer> countMap = new HashMap<>();
+
+		//adds each unique string to HashMap and counts the frequency
+		for(String s : failedLogInUsers){
+			Integer i = countMap.get(s);
+			countMap.put(s, (i == null) ? 1 : i + 1);
+		}
+		return countMap;
+	}
+
+	private void lockUsers(Map<String, Integer> countMap, int maxAttempts){
+
+		//iterates through the map of login frequencies
+		//if user login number>maxAttempts and the account is not already locked, it locks it and prints message
+		for (Map.Entry<String, Integer> set : countMap.entrySet()){
```

Comment:
This `for` loop got a bit messy and hard to read. Since you use some things multiple times, you could have them as variables. Also, try to get into a habit of re-formatting your code regularly :)

Suggestion:

```java
for (Map.Entry<String, Integer> set : countMap.entrySet()) {
        Integer userAttempts = set.getValue();
        String customer = set.getKey();
        List<Customer> customers = bank.getBank().getCustomers();
        
	if (userAttempts >= maxAttempts) {
		if (customers.containsKey(customer) && !isAccountLocked(customer)) {
			bank.getCustomer(customer).setAccountLocked(true);
			out.println("The account: " + customer + " has been locked.\n");
		}
	}
}
```